syntax = "proto3";

package recommender;

service Recommender {
  rpc IsTrained (Empty) returns (IsTrainedResponse);
  rpc GetRecommendations (RecommendationRequest) returns (RecommendationResponse);
  rpc GetGroupRecommendations (GroupRecommendationRequest) returns (GroupRecommendationResponse);
}

message Recommendation {
  int32 recommendedAnimeId = 1;
  int32 rank = 2;
}

message Empty {}

message IsTrainedResponse {
  bool trained = 1;
}

message RecommendationRequest {
  int32 animeId = 1;
  optional int32 amount = 2;
  optional int32 offset = 3;
  repeated int32 excludedAnimeIds = 4;
}


message RecommendationResponse {
  int32 animeId = 1;
  repeated Recommendation recommendations = 2;
}

message GroupRecommendationRequest {
  repeated int32 animeIds = 1;
  optional int32 amount = 2;
  optional int32 offset = 3;
  repeated int32 excludedAnimeIds = 4;
}
message GroupRecommendationResponse {
  repeated Recommendation recommendations = 1;
}

